src/thor/costmatrix.cc

```diff
@@ -923,7 +923,8 @@ void CostMatrix::CheckReverseConnections(const uint32_t target,
       // Special case - common edge for source and target are both initial edges
       if (rev_pred.predecessor() == kInvalidLabel && fwd_predidx == kInvalidLabel) {
         // bail if the edge wasn't allowed
-        if (!rev_pred.path_id()) {
+        if (!fwd_label.path_id()) {
```

Comment:
On the reverse edge label, we store the allowed flag in `path_id` for the _opposing edge_, we should be looking at the forward label's `path_id` here.